src/common/reducers/entities/index.js

```diff
@@ -1,27 +1,43 @@
 import merge from 'lodash/merge';
 
 import * as RepoActionTypes from '../../actions/repository';
-
 import repository from './repository';
 
-
 // TODO snaps
 
-export function entities(state = { snaps: {}, repos: {} }, action) {
+export function entities(state = {
+  snaps: {},
+  repos: {},
+  owners: {}
+}, action) {
   if (action.payload && action.payload.entities) {
     return merge({}, state, action.payload.entities);
   }
 
   // only modify repos if action is one of REPO_ types
-  if (action.payload && action.payload.id && RepoActionTypes[action.type]) {
+  if (RepoActionTypes[action.type]) {
```

Comment:
Maybe we could now move `if (RepoActionTypes[action.type]) {` into `reduceRepoEntity`.

Like:

```js
export function entities(state, action) {
   // do the merge payload.entities merge here

   return {
      ...state,
      ...reduceRepoEntity(state, action)
      // ...reduceSnapEntity(state, action) in future
   }
}

function reduceRepoEntity(state, action) {
   if (RepoActionTypes[action.type]) {
      return reduceEntityHelper(state, action, repository, 'repos');
   }
   return state;
}
```

Maybe even `reduceEntityHelper` could do the conditional thing, but it would make it even more magical.